From: Tanu Kaskinen <tanuk@iki.fi>
Subject: [PATCH] zeroconf-publish: fix unitialized DBusError
Date: Wed, 11 Jan 2017 08:34:27 +0200
Message-Id: <20170111063427.9588-1-tanuk@iki.fi>
X-Mailer: git-send-email 2.11.0

The uninitialized error struct might be the cause for the crash
reported here: https://bugs.archlinux.org/task/52484

---
 src/modules/module-zeroconf-publish.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/modules/module-zeroconf-publish.c b/src/modules/module-zeroconf-publish.c
index e9710292a..4cbfd1eab 100644
--- a/src/modules/module-zeroconf-publish.c
+++ b/src/modules/module-zeroconf-publish.c
@@ -679,6 +679,8 @@ static char *get_icon_name(pa_module*m) {
     DBusConnection *conn = NULL;
     DBusMessageIter sub;
 
+    dbus_error_init(&error);
+
     if (!(bus = pa_dbus_bus_get(m->core, DBUS_BUS_SYSTEM, &error))) {
         pa_log("Failed to get system bus connection: %s", error.message);
         goto out;
-- 
2.11.0

